pkg/utils/annotations/annotations.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/logger"
+)
+
+func GetTargetNameFromAnnotation(
+	logger logger.Logger,
+	logPrefix string,
+	in datasource.FieldAccessor,
+	targetAnnotation string,
```

Comment:
targetAnnotation could be replaced by logPrefix + `.target` (at least for the formatters pkg currently), but I guess you separated it to have the full annotation key (e.g. `timestampTargetAnnotation`) mentioned in code and thus searchable, right?

Reply:
Right, I prefer to keep them separated. I know it's more verbose but also more flexible.